lib/ansible/modules/windows/win_chocolatey.ps1

```diff
@@ -77,7 +77,14 @@ Function Chocolatey-Install-Upgrade
             Fail-Json $result "Chocolatey bootstrap installation failed."
         }
         $result.changed = $true
-        $script:executable = "C:\ProgramData\chocolatey\bin\choco.exe"
+        if (Test-Path "Env:\ChocolateyInstall")
```

Comment:
Looks good, I would actually add a few more checks like the below, feel free to modify it how you would like but this would effectively means it

* check if choco.exe is on the PATH and favour that above all
* check `ChocolatelyInstall`
* check where `ProgramData` is set to, in the rare case C: isn't the main drive
* finally use the hardcoded value it currently has

```
$command = Get-Command -Name "choco.exe"
if ($command) {
    $path = $command.Path
} else {
    $env_value = $env:ChocolateyInstall
    if ($env_value) {
        $path = "$($env:ChocolateyInstall)\bin\choco.exe"
    } else {
        $program_data_env = $env:ProgramData
        if ($program_data_env) {
            $path = "$($env:ProgramData)\chocolatey\bin\choco.exe"
        } else {
            $path = "C:\ProgramData\chocolatey\bin\choco.exe"            
        }
    }
}

if (-not (Test-Path -Path $path) {
    Fail-Json -obj $result -message "failed to find choco.exe, make sure it is added to the PATH or the env var ChocolateyInstall is set"
}
```